SQF-VM/SQF_commands.c

```diff
@@ -3000,79 +2801,376 @@ void CMD_SET(void* input, CPCMD self)
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+	} else {
+		PVALUE val = array_popAt(array, index);
+		push_stack(vm, vm->stack, inst_value(*val));
+	}
+
+	inst_destroy(left);
+	inst_destroy(right);
+}
+
+void CMD_REVERSE(void* input, CPCMD self) {
+	PVM vm = input;
+	PINST right;
+	PVALUE right_val;
+	PARRAY array;
+
+	right = pop_stack(vm, vm->work);
+	right_val = get_value(vm, vm->stack, right);
+
+	if (right_val == 0) {
+		inst_destroy(right);
+		return;
+	}
+
+	if (right_val->type != ARRAY_TYPE()) {
+		vm->error(vm, ERR_RIGHT_TYPE ERR_ARRAY, vm->stack);
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+
+		inst_destroy(right);
+		return;
+	}
+
+	array = right_val->val.ptr;
+
+	array_reverse(array);
+
+	push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+
+	inst_destroy(right);
+}
+
+void CMD_FIND(void* input, CPCMD self) {
+	PVM vm = input;
+	PINST left;
+	PINST right;
+	PVALUE left_val;
+	PVALUE right_val;
+
+	left = pop_stack(vm, vm->work);
+	right = pop_stack(vm, vm->work);
+	left_val = get_value(vm, vm->stack, left);
+	right_val = get_value(vm, vm->stack, right);
+
+	if (left_val == 0 || right_val == 0) {
+		inst_destroy(left);
+		inst_destroy(right);
+		return;
+	}
+
+	if (left_val->type == NOTHING_TYPE() || right_val->type == NOTHING_TYPE()) {
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+
+		inst_destroy(left);
+		inst_destroy(right);
+		return;
+	}
+
+	if (left_val->type != ARRAY_TYPE()) {
+		vm->error(vm, ERR_LEFT_TYPE ERR_ARRAY, vm->stack);
+
+		inst_destroy(left);
+		inst_destroy(right);
+
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+		return;
+	}
+
+	PARRAY array = left_val->val.ptr;
+
+	int index = find(vm, array, right_val);
+
+	push_stack(vm, vm->stack,
+			inst_value(value(SCALAR_TYPE(), base_float(index))));
+
+	inst_destroy(left);
+	inst_destroy(right);
+}
+
+void CMD_APPEND(void* input, CPCMD self) {
+	PVM vm = input;
+	PINST left;
+	PINST right;
+	PVALUE left_val;
+	PVALUE right_val;
+
+	left = pop_stack(vm, vm->work);
+	right = pop_stack(vm, vm->work);
+	left_val = get_value(vm, vm->stack, left);
+	right_val = get_value(vm, vm->stack, right);
+
+	if (left_val == 0 || right_val == 0) {
+		inst_destroy(left);
+		inst_destroy(right);
+		return;
+	}
+
+	if (left_val->type != ARRAY_TYPE()) {
+		vm->error(vm, ERR_LEFT_TYPE ERR_ARRAY, vm->stack);
+
+		inst_destroy(left);
+		inst_destroy(right);
+
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+		return;
+	}
+
+	if (right_val->type != ARRAY_TYPE()) {
+		vm->error(vm, ERR_RIGHT_TYPE ERR_ARRAY, vm->stack);
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+
+		inst_destroy(left);
+		inst_destroy(right);
+
+		return;
+	}
+
+	PARRAY array1 = left_val->val.ptr;
+	PARRAY array2 = right_val->val.ptr;
```

Comment:
Please declare all used variables in the functions begining before first non-declaring line